README.Rmd

```diff
@@ -52,159 +48,63 @@ of common operations:
-with source builds for other platforms.
-
-Binary packages on R-universe are compiled by nightly Rust, with nightly features enabled.
-
-```r
-# Binary installation for x86_64 Windows and macOS, source for other platforms
-install.packages("polars", repos = "https://rpolars.r-universe.dev")
-```
-
-```r
-# Binary installation for Ubuntu 22.04 (x86_64)
-install.packages("polars", repos = "https://rpolars.r-universe.dev/bin/linux/jammy/4.3")
-```
-
-Special thanks to Jeroen Ooms ([@jeroen](https://github.com/jeroen)) for the
-excellent R-universe support.
-
-### GitHub releases
-
-Binary packages on GitHub releases are compiled by nightly Rust, with nightly features enabled.
-
-See latest and all previous [GitHub Releases here](https://github.com/pola-rs/r-polars/releases).
-
-You can download and install these files manually, or install directly
-from R. Simply match the URL for your operating system and the desired release. For example, to
-install the latest release of **polars** on one can use:
-
-Just remember to invoke the `repos = NULL` argument if you are installing these
-binary builds directly from within R.
-
-#### Linux (x86_64)
-
-```r
-install.packages(
-  "https://github.com/pola-rs/r-polars/releases/latest/download/polars__x86_64-pc-linux-gnu.gz",
-  repos = NULL
-)
-```
-
-#### Windows (x86_64)
-
-```r
-install.packages(
-  "https://github.com/pola-rs/r-polars/releases/latest/download/polars.zip",
-  repos = NULL
-)
-```
-
-#### macOS (x86_64)
-
-```r
-install.packages(
-  "https://github.com/pola-rs/r-polars/releases/latest/download/polars__x86_64-apple-darwin20.tgz",
-  repos = NULL
-)
-```
-
-### Build from source
-
-For source installation, pre-built Rust libraries may be available
-if the environment variable `NOT_CRAN` is set to `"true"`. (Or, set `LIBR_POLARS_BUILD` to `"false"`)
+The recommended way to install this package is via R-universe:
 
 ```r
 Sys.setenv(NOT_CRAN = "true")
 install.packages("polars", repos = "https://rpolars.r-universe.dev")
 ```
 
-Otherwise, the Rust library will be built from source.
-the Rust toolchain (Rust `r RcppTOML::parseTOML("src/rust/Cargo.toml")$package$"rust-version"` or later) must be configured.
-
-Please check the <https://github.com/r-rust/hellorust> repository for about Rust code in R packages.
-
-```{r, include = FALSE}
-rust_toolchain_version = read.dcf(
-  "DESCRIPTION",
-  fields = "Config/polars/RustToolchainVersion", all = TRUE
-)[1, 1]
-```
-
-During source installation, some environment variables can be set to enable Rust features and profile changes.
-
-- `RPOLARS_FULL_FEATURES="true"` (Build with nightly feature enabled, requires Rust toolchain `r rust_toolchain_version`)
-- `RPOLARS_PROFILE="release-optimized"` (Build with more optimization)
+Other installation methods and installation details can be
+found in the [Install vignette](https://rpolars.github.io/vignettes/install/)
+(`vignette("install", "polars")`).
```

Comment:
Good point.
However, considering the feature flags, I think there is a possibility that it will be necessary to open this vignette after installation.

Also, there may be a system like pkgdown that provides automatic links to vignette, so I think it is worth including the vignette function.

Reply:
Ok

> there may be a system like pkgdown that provides automatic links to vignette

That might be harder to setup than I thought given that `mkdocs` might override the HTML generated by `downlit`, but that's better discussed in #525

Reply:
I'm not talking about altdoc, just pointing out that this documentation may be used by other static site generators.
For example, I think this is currently being done in R-universe.